hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java

```diff
@@ -42,14 +42,10 @@ class PartitionArrangement {
     PartitionArrangement(Map<Address, int[]> partitionAssignment, Address thisAddress) {
         remotePartitionAssignment = new HashMap<>(partitionAssignment);
         localPartitions = remotePartitionAssignment.remove(thisAddress);
-        int partitionCount = 0;
-        for (int[] value : partitionAssignment.values()) {
-            partitionCount += value.length;
-        }
-        allPartitions = new int[partitionCount];
-        for (int i = 0; i < allPartitions.length; i++) {
-            allPartitions[i] = i;
-        }
+        allPartitions = partitionAssignment.values().stream()
+                .flatMapToInt(a -> Arrays.stream(a))
+                .sorted()
```

Comment:
@Fly-Style `sorted` increases complexity from O(N) to O(NlogN) N - number of partitions. Any comments on that? TDD update? Big clusters have lots of partitions and this change impacts every Jet job.

Reply:
this implementation was quick and simple for demo but also important assumption changes here, `allPartitions` javadoc has to be updated. Also `localPartitions` no longer may contain all local partitions. What is the impact of that?

Reply:
> sorted increases complexity from O(N) to O(NlogN) N - number of partitions. Any comments on that?

You wrote this, can't comment it much. I will remove `sorted`.

Reply:
> Also `localPartitions` no longer may contain all local partitions.

Nothing wrong happens here from "contain **all** local partitions" perspective.

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/96b0110c977e907bf2d85495319989801588cfd2

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/3b523abe5c1abad4fd3a604d3d59006a9f9c4316

Reply:
It seems to me that you avoided analysis of the impact or other solutions, but at least now it does not increase runtime for non-pruned jobs with large partitions numbers.
I will close this comment when this is described in TDD.

Reply:
> I will close this comment when this is described in TDD

It is not related to design, it's implementation detail. Why is it worth describing it in TDD?

Reply:
increased complexity is not a small deal (everything > O(N) should be treated with care). When using member pruning with thousands of partitions this could be a place that limits performace. It should be at least mentioned that some code in case of partition pruning has worse than linear runtime.